src/main/java/cs2103/duke/Ui.java

```diff
@@ -0,0 +1,146 @@
+     */
+    public String showGoodbye() throws IOException {
+        // Goodbye message
+        String goodbye = "Thank you for using Duke: Level " + lv + "\n"
+                + "See you soon!";
+
+        return (sandwich(goodbye));
+    }
+
+    /**
+     * Reads user's inputs and responds to it accordingly.
+     *
+     * @param scanner   The scanner object used to read user inputs.
+     * @param userInput The information entered by the user after the command.
+     * @param tasks     The Tasklist object containing all currently present tasks.
+     * @return The string representing the task added by the user.
+     * @throws DukeException If user enters an invalid input.
+     */
+    public String handleInput(Scanner scanner, String userInput, TaskList tasks) throws DukeException {
+        switch (userInput) {
+        case "list":  // user inputs 'list', return all text stored
+            return tasks.listTasks();
+        case "done":  // first input is done, check second input for integer
+            if (scanner.hasNextInt()) {
+                int taskNum = scanner.nextInt();
+                return tasks.finishTask(taskNum);
+            } else throw new DukeException("unspecified task to mark as done");
+        case "delete":
+            if (scanner.hasNextInt()) {
+                int taskNum = scanner.nextInt();
+                return tasks.deleteTask(taskNum);
+            } else throw new DukeException("unspecified task to delete");
+        case "todo":
+            String todoName = scanner.nextLine();
+            if (todoName.trim().equals("")) {
+                throw new DukeException("No task description");
+            }
+            return tasks.addTask("todo", todoName, "");
+        case "deadline":
+            String[] deadlineTokens = scanner.nextLine().split("\\s*/by\\s*");
+            if (deadlineTokens.length == 0) {
+                throw new DukeException("No task description");
+            } else if (deadlineTokens.length == 1) {
+                throw new DukeException("No task deadline");
+            }
+            String deadlineName = deadlineTokens[0];
+            String deadlineReminder = deadlineTokens[1];
+            return tasks.addTask("deadline", deadlineName, deadlineReminder);
+        case "event":
+            String[] eventTokens = scanner.nextLine().split("\\s*/at\\s*");
+            if (eventTokens.length == 0) {
+                throw new DukeException("No task description");
+            } else if (eventTokens.length == 1) {
+                throw new DukeException("No task duration");
+            }
+            String eventName = eventTokens[0];
+            String eventReminder = eventTokens[1];
+            return tasks.addTask("event", eventName, eventReminder);
+        case "find":
+            if (scanner.hasNext()) {
+                return tasks.findTasks(scanner.nextLine());
+            } else throw new DukeException("unspecified keyword to search for");
```

Comment:
Perhaps you could consider wrapping each case in a private method, for better code readability and easier code management later on.